test_harness/src/lib.rs

```diff
@@ -884,6 +900,396 @@ mod tests {
             .send()
             .await?)
     }
+
+    // Potential TODO's: Merge both test cases into one and find a way to retrieve heartbeats
```

Comment:
You can retrieve the heartbeats using `get_our_latest_msg` after running a single tick.